src/isNonPositive.js

```diff
@@ -0,0 +1,30 @@
+import { both, lte, flip } from 'ramda';
+
+import isNumber from './isNumber';
+
+/**
+ * Checks if value is a non-positive `Number` primitive or object. This includes all negative
+ * numbers and zero.
+ *
+ * @func isNonPositive
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/1.15.0|v1.15.0}
+ * @category Type
+ * @sig * -> Boolean
+ * @param {*} val The value to test
+ * @return {Boolean}
+ * @see {@link RA.isNegative|isNegative}
+ * @example
+ *
+ * RA.isNonPositive(0); // => true
+ * RA.isNonPositive(-1); // => true
+ * RA.isNonPositive(-Infinity); // => true
+ * RA.isNonPositive(Number.MIN_VALUE); // => true
+ *
+ * RA.isNonPositive(Infinity); // => false
+ * RA.isNonPositive(Number.MAX_VALUE); // => false
+ * RA.isNonPositive(NaN); // => false
+ */
+const isNonPositive = both(isNumber, flip(lte)(0));
```

Comment:
Would this be better for composition ?
```js
const isNonPositive = both(isNumber, complement(isPositive));
```

Reply:
Does that make sense though given that `isPositive` is also checking for `isNumber`? I was thinking that this might complicate things, but I guess we're rechecking `isNumber` anyway. However this would mean we are checking it twice.

Reply:
I always prefer composition even though It may not be the most effective thing how to do things. We can always optimize later. I consider checking `isNumber` in `isPositive` an implementation detail in this case.

Reply:
OK. I'll buy that for a dollar.

Reply:
@char0n My original intuition was correct. The problem comes with `NaN`:

```
const isNonPositive = both(isNumber, complement(isPositive));
isNonPositive(NaN) // true
```

`NaN` is neither positive or non-positive, so this should be `false` shouldn't it? Using `complement` we are effectively doing `!both(isNumber, isPositive)`,  so NaN passes `isNumber` but fails `isPositive` giving a `false`, however the complement is `true`.

Reply:
Yeah, so the question is - should we consider NaN a non-positive number ? IMHO, I think no (but that is debatable). By this assumption we call use your original signature that seems to avoid NaN problem. But I would add note to function description that NaN is not considered non-positive number.

Reply:
I think this is the most practical solution.